engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobBatchCollector.java

```diff
@@ -128,15 +128,15 @@ Either<TooLargeJob, Integer> collectJobs(final TypedRecord<JobBatchRecord> recor
   private void appendJobToBatch(
       final ValueArray<JobRecord> jobIterator,
       final ValueArray<LongValue> jobKeyIterator,
-      final ExpandableArrayBuffer jobCopyBuffer,
-      final Long key,
+      final long key,
       final JobRecord jobRecord) {
     jobKeyIterator.add().setValue(key);
     final JobRecord arrayValueJob = jobIterator.add();
 
     // clone job record since buffer is reused during iteration
+    final var jobCopyBuffer = new UnsafeBuffer(ByteBuffer.allocate(jobRecord.getLength()));
```

Comment:
Required since adding a new job to the batch does not automatically "flush" the previous one as before. Previously, it would serialize the last written job, thus copying it over to the array's underlying buffer - thus we could reuse the same allocation buffer here.

Reply:
I find this a bit complex to understand or event catch this, if I would miss this we would reuse the object again. Sounds like an issue we had earlier as well, which we run into quite often. Can we find a solution for this as well?

Reply:
We could add a `BaseValue#copy` method analogous `wrap` but which performs a full copy of the object instead.

I don't know of a way to completely prevent it though - the callee (in this case, the ArrayValue#wrap`) can't really know whether it has to copy, or if it's safe to just wrap, so the caller has to do it.

We were doing it here, after all - it's just we were copying it once in a buffer, then copying it _again_ (but hidden from the caller) into another buffer in the `ArrayValue`. Now it's just more explicit, I guess...